_std/macros/transfers.dm

```diff
@@ -0,0 +1,4 @@
+/// Transfer signal helpers
+
+/// Transfer the given movable to the output of the given target or to the loc of the target.
+#define TRANSFER_OR_DROP(T, AM) if (!SEND_SIGNAL(T, COMSIG_TRANSFER_OUTGOING, AM)){AM.set_loc(T.loc)}
```

Comment:
Hm, where do you expect this macro to be used? Presumably in internal transfer with fixed target (i.e. not ones where the player can choose the target), right?

Reply:
This is a common pattern in all machines. The machine has to send an outgoing transfer signal for the component to
handle the output targeting. If it returns false, then the machine has to figure out itself where to put it. The component doesn't assume because some machines may want to keep failed outputs inside of themselves, so this macro exists.

Reply:
Oh, that's what confused me. I thought `T` was the target, silly me haha.